pkg/utils/generate.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type PodArg struct {
+	Name            string
+	Status          v1.PodPhase
+	Namespace       string
+	Ans             map[string]string
+	Labels          map[string]string
+	ContainerStatus v1.ContainerStatus
+	Nodename        string
+}
+
+func newPod(p PodArg) v1.Pod {
+	if p.Status == "" {
+		p.Status = v1.PodRunning
+	}
+	if p.Namespace == "" {
+		p.Namespace = metav1.NamespaceDefault
+	}
```

Comment:
I think we could use default value for values that we don't care in certain test. 
When we need them, for example
https://github.com/chaos-mesh/chaos-mesh/blob/e63c877a5503b990a48b2c239847047677cba94e/pkg/utils/selector_test.go#L333-L336
When we explicitly compare pod status, we could still write them out to clarify what we test.

Reply:
One thing I'm not so sure about the constructor part. Do we need 
```go
func newPodArg() *PodArg {
    p := new(PodArg)
    p.Status = v1.PodRunning
    return p
}
```
or
```go
func newPodArg(status v1.PodPhase) *PodArg {
    p := new(PodArg)
    if status == nil {
        p.Status = v1.PodRunning
    }
    return p
}
```

Reply:
Just creating each constructors that ALL you need. 😆
As go do not support method overloading like java, so you should use different method names for the different constructors.

Reply:
Wait do you mean I should create `newPodArg`, `newPodArgForRunning`, `newPodArgForDefaultNs` and `newPodArgForRunningAndDefaultNs`? Yes it could explicitly declare the pod status, but it's ... just a bit weird to me... I'd rather still expose them as arguments in this case 🤪

But I still believe it is acceptable not explicitly declaring arguments that not matter. What do you think about it

Reply:
TLDR; You are right, it is acceptable. 🙈

It's also indeed weird to me. And I could not provide a perfect solution. 🙈 

I could share my thought about this situation:

- it's test cases, so I would like to claim all states explicitly as possible and make comments on which state will affect the logic.
- then I consider the fluent codes, the "builder pattern"`NewDefaultBuilder().SetPhase().SetNs().Build()` is the best choice for one-line construction, but it will take a lot of codes to create a builder. So I prefer just write a plain `Struct`  with all fields to here.

When another developer wants to understand these codes, he/she might feel confusing about the empty field of the struct, until he/she find the default value logic in `newPods()`.

Reply:
Yeah I can see that, cause I'm also a bit unsure about not explicitly declare everything, but the attraction of making code clean is just so much... I tried to add tags and comments to `PodArg` to declare default values are set. It might be of help to other developers 🙈